rust/src/http2/http2.rs

```diff
@@ -0,0 +1,892 @@
+/* Copyright (C) 2020 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+use super::files::*;
+use super::parser;
+use crate::applayer::{self, *};
+use crate::core::{
+    self, AppProto, Flow, SuricataFileContext, ALPROTO_FAILED, ALPROTO_UNKNOWN, IPPROTO_TCP,
+    STREAM_TOCLIENT, STREAM_TOSERVER,
+};
+use crate::filecontainer::*;
+use crate::filetracker::*;
+use crate::log::*;
+use nom;
+use std;
+use std::ffi::{CStr, CString};
+use std::mem::transmute;
+
+static mut ALPROTO_HTTP2: AppProto = ALPROTO_UNKNOWN;
+
+const HTTP2_DEFAULT_MAX_FRAME_SIZE: u32 = 16384;
+
+//TODOask why option ?
```

Comment:
Why is this field an `Option` when it is always `Some` during runtime ?

Reply:
Rust limitation.  Its set from C during initialization, and has to be something before.  There's probably other ways to do this, but when parsers were split a little more over C and Rust, this was best.  If the object is not needed in C at all, maybe it can be initialized in Rust, allowing us to get rid of this.

Reply:
Ok thanks, I will try to initialize it from Rust

Reply:
It looks like C `StreamingBufferConfig` is not shared with Rust `SuricataStreamingBufferConfig`

I mentioned this in the rustily everything ticket https://redmine.openinfosecfoundation.org/issues/3195